apis/v1alpha1/route_types.go

```diff
@@ -35,6 +35,24 @@ const (
 	GatewayAllowSameNamespace GatewayAllowType = "SameNamespace"
 )
 
+const (
+	// AnnotationAppProtocol defines the protocol a Gateway should use for
+	// communication with a Kubernetes Service.
+	// This annotation must be applied to the `BackendPolicy` resource and will
+	// apply to all Service resources that are selected by
+	// `BackendPolicy.Spec.BackendRefs`. The annotation serves as a stop-gap
+	// solution for the Kubernetes versions < 1.18, where the `AppProtocol`
+	// field on the `Service` resource is not available. Using this annotation
+	// is highly discouraged when AppProtocol field is available. The AppProtocol
+	// field, when populated, takes precedence over this annotation.
+	//
+	// Examples:
+	//
+	// - `networking.x-k8s.io/app-protocol: https`
+	// - `networking.x-k8s.io/app-protocol: tls`
```

Comment:
Do we need a more stringent spec here that matches the `AppProtocol` description as far as protocol names + format supported?

Reply:
Given that we are using this annotation as a stop-gap, should we avoid doing it?
Or is this an API that we think is a stop-gap but will eventually stay around for a long-time?

Defining an unclear behavior gives future ourselves another reason to remove it, but then it also has the potential to confuse users. I don't think users would be confused here though.

Reply:
I think it would be good just to say that any values set for this annotation should also pass validation for the `AppProtocol` field (or something along those lines). I agree that does seem relatively obvious, but I just want to be as clear as possible with the spec here.

Reply:
Added in, PTAL. Thanks!